lib/ts/recipe/passkey/types.ts

```diff
@@ -0,0 +1,473 @@
+    relyingPartyName: (input: { request: BaseRequest | undefined; userContext: UserContext }) => string; // should return the app name
+    getEmailDeliveryConfig: (
+        isInServerlessEnv: boolean
+    ) => EmailDeliveryTypeInputWithService<TypeEmailPasswordEmailDeliveryInput>;
+    override: {
+        functions: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+        apis: (originalImplementation: APIInterface, builder?: OverrideableBuilder<APIInterface>) => APIInterface;
+    };
+};
+
+export type TypeInput = {
+    emailDelivery?: EmailDeliveryTypeInput<TypeEmailPasswordEmailDeliveryInput>;
+    validateEmail?: (value: any, tenantId: string, userContext: UserContext) => Promise<string | undefined>;
+    relyingPartyId?: string | ((input: { request: BaseRequest | undefined; userContext: UserContext }) => string);
+    relyingPartyName?: string | ((input: { request: BaseRequest | undefined; userContext: UserContext }) => string);
+    override?: {
+        functions?: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+        apis?: (originalImplementation: APIInterface, builder?: OverrideableBuilder<APIInterface>) => APIInterface;
+    };
+};
+
+export type RecipeInterface = {
+    registerPasskeyOptions(input: {
+        email: string;
+        password: string;
+        session: SessionContainerInterface | undefined;
+        tenantId: string;
+        userContext: UserContext;
+    }): Promise<{
+        status: "OK";
+        passkeyGeneratedOptionsId: string;
+        rp: {
+            id: string;
+            name: string;
+        };
+        user: {
+            id: string;
+            name: string;
+            displayName: string;
+        };
+        challenge: string;
+        timeout: number;
+        excludeCredentials: {
+            id: string;
+            type: string;
+            transports: ("ble" | "hybrid" | "internal" | "nfc" | "usb")[];
+        }[];
+        attestation: "none" | "indirect" | "direct" | "enterprise";
+        pubKeyCredParams: {
+            alg: number;
+            type: string;
+        }[];
+        authenticatorSelection: {
+            requireResidentKey: boolean;
+            residentKey: "required" | "preferred" | "discouraged";
+            userVerification: "required" | "preferred" | "discouraged";
+        };
+    }>;
+
+    signInPasskeyOptions(input: {
+        session: SessionContainerInterface | undefined;
+        tenantId: string;
+        userContext: UserContext;
+    }): Promise<{
+        status: "OK";
+        passkeyGeneratedOptionsId: string;
+        challenge: string;
+        timeout: number;
+        userVerification: "required" | "preferred" | "discouraged";
+    }>;
+
+    signUp(input: {
+        email: string | undefined;
```

Comment:
do we need an email here? Given that the first step in signing up is to generate a passkeyoptions, and that function already takes in an email?